src/clarity.rs

```diff
@@ -200,6 +207,44 @@ where F: FnOnce(MarfedKV) -> (MarfedKV, R) {
     result
 }
 
+fn eval(invoked_by: &str, args: &[String]) -> EvalInput {
```

Comment:
Can you change this method name to something like `parse_eval()`?  `eval()` is already used in other modules.